### DISCUSSION TITLE
Total Work Item Age Projections

### DISCUSSION THREAD

**Discussion by huserben** [nbrown02/ADO-Process-Behaviour-Chart#4] [Ideas]
Hi @nbrown02 

After some discussions in the office as well as on [Slack](https://prokanban.slack.com/archives/C01846C0PKK/p1705514747812869), I wanted to try out if we can use Total Work Item Age (TWIA) for limiting WIP.

The first idea was to simply "project" the TWIA to the next days assuming the WIP stays the same. I somehow managed this based on the existing template:

![TWIA Projection](https://github.com/nbrown02/Process-Behaviour-Chart/assets/5486874/7fb7006a-5722-4088-a08d-eb815282e794)

I struggled a bit showing "future" dates as I think it's all connected to the slicer on top (so I just copy pasted it into a new page ;-)).

I'm also thinking, maybe it could be made somehow configurable. Like start out with the "current WIP", but be able to reduce or increase and see what will happen with TWIA to make decisions on whether new work should be pulled in.

Few questions:
- Do you think something like this is possible? If so, I might need some input on how to create this to be more configurable for the users
- Do you think it would make sense to have this as part of the PBC template of yours?

Cheers
Benji

**Comment by nbrown02** [comment]
Anything is possible ;)So for now it’s something I wouldn’t consider for the main template but you could fork and add a link to your forked version in the show and tell part of the discussion page?You’re right on the slicer, there is an if statement filtering it :)On 26 Jan 2024, at 09:03, Benjamin Huser-Berta ***@***.***> wrote:﻿
Hi @nbrown02
After some discussions in the office as well as on Slack, I wanted to try out if we can use Total Work Item Age (TWIA) for limiting WIP.
The first idea was to simply "project" the TWIA to the next days assuming the WIP stays the same. I somehow managed this based on the existing template:
image.png (view on web)
I struggled a bit showing "future" dates as I think it's all connected to the slicer on top (so I just copy pasted it into a new page ;-)).
I'm also thinking, maybe it could be made somehow configurable. Like start out with the "current WIP", but be able to reduce or increase and see what will happen with TWIA to make decisions on whether new work should be pulled in.
Few questions:

Do you think something like this is possible? If so, I might need some input on how to create this to be more configurable for the users
Do you think it would make sense to have this as part of the PBC template of yours?

Cheers
Benji

**Reply by huserben** [reply]
Yeah I thought it might be a bit "special interest" and doesn't fit ideally into this template.

I'll play around with it and once I have a decent version that we also tried out in real life, I'll update you here and we could add a link :)

**Reply by huserben** [reply]
Yesterday we agreed on the first version of what we want to use:
![TWIA Projection](https://github.com/nbrown02/Process-Behaviour-Chart/assets/5486874/88f32134-7358-41a9-849c-da9afe667153)

I've created a new template in a new repo: [Total-Work-Item-Age-Projection-Power-BI-Template](https://github.com/huserben/Total-Work-Item-Age-Projection-Power-BI-Template)

It's very raw and there is no proper description etc. yet. The plan is to use this template for a bit and see what we want/need to change to make it more usable for controlling WIP.

I'll keep you posted on the progress after we gathered some real-life experience with it 😉

**Comment by nbrown02** [comment]
Love it!On 27 Jan 2024, at 11:36, Benjamin Huser-Berta ***@***.***> wrote:﻿
Yesterday we agreed on the first version of what we want to use:
image.png (view on web)
I've created a new template in a new repo: Total-Work-Item-Age-Projection-Power-BI-Template
It's very raw and there is no proper description etc. yet. The plan is to use this template for a bit and see what we want/need to change to make it more usable for controlling WIP.
I'll keep you posted on the progress after we gathered some real-life experience with it 😉

**Comment by nbrown02** [comment]
@huserben could you add some more detail in https://github.com/nbrown02/Process-Behaviour-Chart/discussions/categories/show-and-tell when you get a chance? No rush of course :)

**Reply by huserben** [reply]
Yeah will do. Do you want me to add something right away or should we wait a bit till we have the template "battle-tested"? Or shall I just add the reference here and more details can be found on the GitHub repo with the template anyway?

**Comment by nbrown02** [comment]
Entirely up to you :)Sent from my iPhoneOn 29 Jan 2024, at 16:35, Benjamin Huser-Berta ***@***.***> wrote:﻿
Yeah will do. Do you want me to add something right away or should we wait a bit till we have the template "battle-tested"? Or shall I just add the reference here and more details can be found on the GitHub repo with the template anyway?

**Reply by huserben** [reply]
Done. No need to keep this task open longer than it needs to be 😉
